spec/features/merchants/show_spec.rb

```diff
@@ -30,4 +30,72 @@
       expect(current_path).to_not eq("/merchants/#{merchant_2.id}/invoices")
     end
   end
+
+  describe "Merchant Dashboard Stats - Top 5 Customers" do
+    let!(:merchant_1) { Merchant.create!(name: "Steve's Soaps") }
+    let!(:merchant_2) { Merchant.create!(name: "Charlie's Chia Pets") }
+    let!(:customer_1) { create(:customer) } # 5 successful transactions
+    let!(:customer_2) { create(:customer) } # 4 successful transactions
+    let!(:customer_3) { create(:customer) } # 3 successful transactions
+    let!(:customer_4) { create(:customer) } # 2 successful transactions
+    let!(:customer_5) { create(:customer) } # 1 successful transactions
+    let!(:customer_6) { create(:customer) } # 0 successful transactions
+    
+    before(:each) do
+      5.times do
+        invoice = create(:invoice, customer: customer_1)
+        create(:transaction, result: true, invoice: invoice)
+      end
+
+      4.times do
+        invoice = create(:invoice, customer: customer_2)
+        create(:transaction, result: true, invoice: invoice)
+      end
+
+      3.times do
+        invoice = create(:invoice, customer: customer_3)
+        create(:transaction, result: true, invoice: invoice)
+      end
+
+      2.times do
+        invoice = create(:invoice, customer: customer_4)
+        create(:transaction, result: true, invoice: invoice)
+      end
+
+      invoice = create(:invoice, customer: customer_5)
+      create(:transaction, result: true, invoice: invoice)
+      create(:transaction, result: false, invoice: invoice)
+
+      2.times do
+        invoice = create(:invoice, customer: customer_6)
+        create(:transaction, result: false, invoice: invoice)
+      end
+    end
+    
+    it "display names of top 5 customers with largest successful transactions " do
+    
+      visit "/merchants/#{merchant_1.id}/dashboard"
+
+      within("#top5") do
+      expect(customer_1.first_name).to appear_before(customer_2.first_name)
```

Comment:
tiny thing that you don't have to change but can just note for future -- for within blocks, it's helpful to indent the expect statements so we can visually know to expect those assertions within that specific block in the view.